core/execution/special_orders.go

```diff
@@ -410,15 +410,20 @@ func (m *Market) applyBondPenaltiesAndLiquidationExcludingPending(
 		// now we can had them to the cancelled map
 		// as we don't need to use it anymore apart for returning
 		for _, v := range reallyClosed {
+			party := v.Party()
+			if m.liquidity.IsLiquidityProvider(party) {
+				err := m.cancelLiquidityProvision(ctx, party, false)
+				if err != nil {
+					m.log.Error("Failed to cancel LP",
+						logging.Error(err),
+						logging.PartyID(party))
+				}
+			} else {
+				// party not an LP, cancel only their pegged orders at this point
+				m.CancelAllOrders(ctx, party, true)
```

Comment:
Needed to prevent a regression (there was a unit test failing without, added for this issue https://github.com/vegaprotocol/vega/issues/2772). It's because the LP and party with a pegged order go through the same flow here. Cancelling the pegged orders, rest of the closeout will be handled at MTM time, just like for LPs